src/com/facebook/buck/go/GoBuckConfig.java

```diff
@@ -109,7 +110,11 @@ Path getDefaultPackageName(BuildTarget target) {
       public Tool get() {
         Optional<Path> toolPath = delegate.getPath("go", configName);
         if (!toolPath.isPresent()) {
-          toolPath = Optional.of(goToolDirSupplier.get().resolve(toolName));
+          String exeName = toolName;
+          if (delegate.getPlatform() == Platform.WINDOWS) {
+            exeName = toolName + ".exe";
+          }
```

Comment:
Hmm AFAICT `ExecutableFinder` might need some changes to respect "in this path", since the binaries here are called something very generic (`compile` and `link`), and should not be looked up in $PATH.

I may just revert this change, since go tests still fail due to lack of C linker. Go technically doesn't need a C linker, but it doesn't seem like there's a way to check if the linker is "real" or not.